Lesson2

MikhailFilippov1 opened the pull request.

Commit:
Домашнее задание №1. Правильно ли я воспользовался возможностями interface?

Commit:
ДЗ №2. Честно говоря, я не знаю, как вы проверяете ДЗ: смотрите код или просто запускаете RUN. Если RUN, то я попытался сделать какой-никакой интерфейс, чтобы удобнее было проверять. Теперь вопрос:
Почему я не могу в эту конструкцию
catch (NumberFormatException e) {
                    System.out.printf("Can't parse String %s into Integer in the [%d][%d] cell\n", array[y][x], y, x);
 return;
 } вместо NumberFormatException вставить MyArrayDataException? Когда я вставляю свое исключение, программа его не ловит, а выдает результат работы NumberFormatException. Заранее благодарю за ответ.

dark-savant-of-guardia commented on src/main/java/ru/geekbrains/homework_java2/lesson2/HomeWork2.java:
```diff
@@ -0,0 +1,119 @@
+                throw new MyArraySizeException("Size Y more than 4");
+            }
+            for (int x = 0; x < array[y].length; x++) {
+                if (array[y].length > 4) {
```
почему только больше? Вдруг меньше будет

dark-savant-of-guardia commented on src/main/java/ru/geekbrains/homework_java2/lesson2/HomeWork2.java:
```diff
@@ -0,0 +1,119 @@
+                try {
+                    sum += Integer.parseInt(array[y][x].trim());
+                }catch (NumberFormatException e) {
+                    System.out.printf("Can't parse String %s into Integer in the [%d][%d] cell\n", array[y][x], y, x);
+                    return;
```
а тут надо было же исключение тоже выкинуть. Второе

Commit:
ДЗ №2. Честно говоря, я не знаю, как вы проверяете ДЗ: смотрите код или просто запускаете RUN. Если RUN, то я попытался сделать какой-никакой интерфейс, чтобы удобнее было проверять. Теперь вопрос:
Почему я не могу в эту конструкцию
catch (NumberFormatException e) {
                    System.out.printf("Can't parse String %s into Integer in the [%d][%d] cell\n", array[y][x], y, x);
 return;
 } вместо NumberFormatException вставить MyArrayDataException? Когда я вставляю свое исключение, программа его не ловит, а выдает результат работы NumberFormatException. Заранее благодарю за ответ.
